### PR TITLE
Split ContractRequest schema

### PR DESCRIPTION
**Pull request by kiurchv**

### CODE REVIEW

File: apps/graphql/features/contexts/context.exs [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+
+    %{conn: ConnTest.build_conn()}
+  end)
+
+  scenario_finalize(fn status, state ->
+    Sandbox.checkin(Repo)
+    Sandbox.checkin(PRMRepo)
+    Sandbox.checkin(EventManagerRepo)
+
+    with {:ok, _} <- status, do: Mox.verify!()
```
Comment (progsmile) [inline apps/graphql/features/contexts/context.exs]: use if

File: apps/graphql/features/contexts/context.exs [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+             )
+
+      {:ok, state}
+    end
+  )
+
+  then_(
+    ~r/^I should receive collection with (?<count>\d+) items?$/,
+    fn %{resp_entities: resp_entities} = state, %{count: count} ->
+      assert resp_entities
```
Comment (progsmile) [inline apps/graphql/features/contexts/context.exs]: What does `resp_entities` contain in not found case? Suppose should receive empty list

File: apps/graphql/features/contexts/context.exs [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+  then_(
+    ~r/^the (?<field>[\w_]+) of the first item in the collection should be "(?<value>[^"]+)"$/,
+    fn %{resp_entities: resp_entities} = state, %{field: field, value: value} ->
+      resp_value =
+        resp_entities
+        |> hd()
+        |> Map.get(field)
+        |> to_string()
+
+      assert value = resp_value
```
Comment (progsmile) [inline apps/graphql/features/contexts/context.exs]: use ==

File: apps/graphql/features/contexts/context.exs [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+  def put_client_id(conn, id \\ UUID.generate()) do
+    metadata = Jason.encode!(%{"client_id" => id})
+    put_req_header(conn, @consumer_metadata_header, metadata)
+  end
+
+  def post_query(conn, query, variables \\ %{}) do
+    post(conn, @graphql_path, %{query: query, variables: variables})
+  end
+
+  def prepare_attrs(queryable, attrs) when is_map(attrs), do: prepare_attrs(queryable, Map.to_list(attrs))
```
Comment (progsmile) [inline apps/graphql/features/contexts/context.exs]: In codebase most time `params` variable is mentioned, let's use such name

File: apps/graphql/features/contexts/context.exs (hunk 2 above)
Comment (kiurchv) [reply apps/graphql/features/contexts/context.exs]: This step definition checks collection presence and length, in the case when we expect to receive no collection at all we use another condition, see here: https://github.com/edenlabllc/ehealth.api/pull/3711/files#diff-fcb81f96426ffd0a0f94a13fdce6af56R239

File: apps/graphql/lib/graphql_web/schema/contract_request_types.ex [hunk 5]
```diff
@@ -181,48 +116,51 @@ defmodule GraphQLWeb.Schema.ContractRequestTypes do
...
 
-  node object(:contract_request) do
+  interface :contract_request do
+    field(:id, non_null(:id))
     field(:database_id, non_null(:id))
     field(:contract_number, :string)
-    field(:parent_contract, :contract, resolve: load_by_parent(PRM, CapitationContract))
-    field(:previous_request, :contract_request, resolve: dataloader(IL))
-    field(:assignee, :employee, resolve: load_by_parent(PRM, Employee))
+    # field(:parent_contract, :contract)
```
Comment (progsmile) [inline apps/graphql/lib/graphql_web/schema/contract_request_types.ex]: remove

File: apps/graphql/features/capitation_contract_requests/get_all.feature [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+  Scenario: Request all items with NHS client
+    Given there are 2 capitation contract requests exist
+    And my scope is "contract_request:read"
+    And my client type is "NHS"
+    When I request first 10 capitation contract requests
+    Then no errors should be returned
+    And I should receive collection with 2 items
+
+  Scenario: Request belonging items with MSP client
```
Comment (progsmile) [inline apps/graphql/features/capitation_contract_requests/get_all.feature]: 👍👍

File: apps/graphql/features/contexts/context.exs (hunk 1 above)
Comment (kiurchv) [reply apps/graphql/features/contexts/context.exs]: We cannot use match operator in `if` there, so I'd decided to use `with` instead

File: apps/graphql/features/contexts/context.exs (hunk 3 above)
Comment (kiurchv) [reply apps/graphql/features/contexts/context.exs]: For what purpose?

File: apps/graphql/features/contexts/context.exs (hunk 1 above)
Comment (progsmile) [reply apps/graphql/features/contexts/context.exs]: if match?({:ok, _}, status) do ..

File: apps/graphql/features/contexts/context.exs (hunk 4 above)
Comment (kiurchv) [reply apps/graphql/features/contexts/context.exs]: Nope, when inserting a new record to DB we are passing its fields (or attributes). Params are related only to HTTP requests

File: apps/graphql/features/contexts/context.exs (hunk 3 above)
Comment (progsmile) [reply apps/graphql/features/contexts/context.exs]: to be more explicit

File: apps/graphql/lib/graphql_web/schema/contract_request_types.ex (hunk 5 above)
Comment (kiurchv) [reply apps/graphql/lib/graphql_web/schema/contract_request_types.ex]: No, we still need this field, I'll uncomment this when Contract schema will be ready

File: apps/graphql/features/contexts/context.exs (hunk 1 above)
Comment (kiurchv) [reply apps/graphql/features/contexts/context.exs]: Yes, but `with` does the same under the hood

File: apps/graphql/features/contexts/context.exs [hunk 7]
```diff
@@ -0,0 +1,355 @@
...
+      for row <- table_data do
+        attrs = prepare_attrs(LegalEntity, row)
+        insert(:prm, :legal_entity, attrs)
+      end
+
+      {:ok, state}
+    end
+  )
+
+  when_(
```
Comment (PavelVesnin) [inline apps/graphql/features/contexts/context.exs]: suggest to move queries to another file

Comment (kiurchv) [reply apps/graphql/features/contexts/context.exs]: I've tried several approaches to organize step definitions better but ended with this one global context file with all step definitions. Currently, development of our BDD testing infrastructure is on the very early stage. For now, I want to write more tests and make its refactoring later.

File: apps/graphql/lib/graphql_web/schema/contract_request_types.ex (hunk 5 above)
Comment (progsmile) [reply apps/graphql/lib/graphql_web/schema/contract_request_types.ex]: what about next line?
# field(:previous_request, :contract_request)

**Comment by kiurchv** [comment]
@danakonovalenko @IrinaRevutska Since this PR introduces BDD approach on eHealth, I'd like to hear your opinions on Gherkin feature files

File: apps/graphql/lib/graphql_web/schema/contract_request_types.ex (hunk 5 above)
Comment (kiurchv) [reply apps/graphql/lib/graphql_web/schema/contract_request_types.ex]: It seems that previous line no more needed, agree

File: apps/graphql/lib/graphql_web/schema/contract_request_types.ex [hunk 8]
```diff
@@ -114,7 +49,7 @@ defmodule GraphQLWeb.Schema.ContractRequestTypes do
 
     payload field(:approve_contract_request) do
       meta(:scope, ~w(contract_request:update))
-      middleware(ParseIDs, id: :contract_request)
+      middleware(ParseIDs, id: :capitation_contract_request)
```
Comment (progsmile) [inline apps/graphql/lib/graphql_web/schema/contract_request_types.ex]: Why there is :capitation_contract_request ?

**Pull request merged**